test/cases/test_auth.py

```
@@ -0,0 +1,351 @@
+            modified = original_yaml.replace("admin_password: admin42",
+                                             "admin_password: changed123")
+            modified = modified.replace("readonly_password: readonly42",
+                                        "readonly_password: changed456")
+            self.s.post_form("/settings/save/", {"settings": modified})
+
+            resp2 = self.s.session.get(f"{BASE_URL}/login", timeout=10)
+            self.assertNotIn("Default credentials detected", resp2.text)
+        finally:
+            # Visiting /login clears session; re-login with new creds to restore
+            restore = ProxyWebSession()
+            restore.login(username="admin", password="changed123")
```

Ruff (0.15.10)

[error] Possible hardcoded password assigned to argument: "password"

(S106)